cmake/tpl/FindKLU.cmake

```diff
@@ -30,6 +30,18 @@
 #   KLU_LIBRARIES   - all of the libraries needed for KLU
 # ---------------------------------------------------------------
 
+if (NOT (KLU_INCLUDE_DIR OR KLU_LIBRARY_DIR OR KLU_LIBRARY))
+  # prefer import target from upstream SuiteSparse if available and
+  # user didn't point to a specific (different) version
+  find_package(KLU CONFIG)
+
+  if(TARGET SuiteSparse::KLU)
+    set(KLU_FOUND ON)
```

Comment:
Setting `KLU_FOUND` to `ON` manually is only needed because of a bug in SuiteSparse 7.4.0. It should be fixed in SuiteSparse 7.5.0 (which will probably be released shortly).